dev/set_matrix.py

```diff
@@ -0,0 +1,427 @@
+    flavors = set(x.split("-")[0] for x in config.keys())
+    changed_flavors = get_changed_flavors(args.changed_files, flavors)
+
+    job_names = []
+    includes = []
+    should_include_all_items = not args.diff_only
+
+    for flavor, cfgs in config.items():
+        package_info = cfgs.pop("package_info")
+
+        should_include_all_items_in_this_flavor = (
+            should_include_all_items
+            or any(flavor.startswith(x) for x in changed_flavors)
+            or (flavor not in config_ref)
+            or (package_info != config_ref[flavor]["package_info"])
+        )
+
+        for key, cfg in cfgs.items():
+            if (
+                should_include_all_items_in_this_flavor
+                or (key not in config_ref[flavor])
+                or (cfg != config_ref[flavor][key])
+            ):
+                print("Processing {}.{}".format(flavor, key))
+
+                # released versions
+                versions = get_released_versions(package_info["pip_release"])
+                versions = filter_versions(
+                    versions, cfg["minimum"], cfg["maximum"], cfg.get("unsupported"),
+                )
+                versions = select_latest_micro_versions(versions)
+                for ver in versions:
+                    job_name = " / ".join([flavor, ver, key])
+                    job_names.append(job_name)
+                    requirements = process_requirements(cfg.get("requirements"), ver)
+                    install = "pip install -U '{}=={}'".format(package_info["pip_release"], ver)
+                    run = remove_comments(cfg["run"])
+                    includes.append(
+                        {
+                            "job_name": job_name,
+                            "requirements": requirements or None,
+                            "install": install,
+                            "run": run,
+                        }
+                    )
+
+                # development version
+                if "install_dev" in package_info:
+                    job_name = " / ".join([flavor, "dev", key])
+                    job_names.append(job_name)
+                    requirements = process_requirements(cfg.get("requirements"), "dev")
+                    install = remove_comments(package_info["install_dev"])
+                    run = remove_comments(cfg["run"])
+                    includes.append(
+                        {
+                            "job_name": job_name,
+                            "requirements": requirements or None,
+                            "install": install,
+                            "run": run,
+                        }
+                    )
+
+    matrix = {"job_name": job_names, "include": includes}
+    print(divider("Result"))
+    # specify `indent` to prettify the output
+    print(json.dumps(matrix, indent=2))
+
+    if "GITHUB_ACTIONS" in os.environ:
+        # this line prints out nothing in the console on github actions
```

Comment:
Why does it print nothing? Do we need to print nothing?

Reply:
<img width="608" alt="Screen Shot 2020-12-02 at 17 02 01" src="https://user-images.githubusercontent.com/17039389/100845166-24c22300-34c0-11eb-8f20-2026955d8164.png">

This is how it looks like

Reply:
It seems GitHub automatically hide `::set-output:xxx...`

Reply:
I see! Can we update the comment to indicate that this is used to set the output for the Github Actions job but doesn't actually print anything to the console?

Reply:
sure!